test/unit/native_deps/alloc_shims.c

```diff
@@ -0,0 +1,24 @@
+#include <stddef.h>
+#include <stdlib.h>
+
+void* __rdl_alloc(size_t, size_t);
```

Comment:
I'd put this file into generated rust toolchains for linux when cc rules detected clang, but let's do it in a followup PR.